### PR TITLE
Feature/libravatar fallback

### PR DESCRIPTION
**Pull request by Gyran**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -99,7 +99,9 @@ app.get( '/avatar/:emailHash', async ( request, response ) => {
         return false;
     }
 
-    if ( request.params.emailHash.length !== 32 && request.params.emailHash.length !== 64 ) {
+    const emailHash = request.params.emailHash.toLocaleLowerCase();
```
Comment (kokarn) [inline server.js]: Is this spec compliant?

Comment (Gyran) [reply server.js]: where did you find that?

Comment (Gyran) [reply server.js]: forgot to push :(

Comment (kokarn) [reply server.js]: No, I'm just curious what the spec says about lowercase & uppercase

File: server.js [hunk 2]
```diff
@@ -146,7 +141,8 @@ app.get( '/avatar/:emailHash', async ( request, response ) => {
     if ( !forceDefault ) {
         let i = 0;
         while (i < providersOrder.length && !avatarImage) {
-            avatarImage = await providersOrder[i](emailHash, targetSize);
+            const provider = providers[providersOrder[i]];
+            avatarImage = await provider(emailHash, targetSize);
             i = i + 1;
         }
```
Comment (kokarn) [inline server.js]: Nitpicking ofc but I _feel_ something like this is a bit more readable?
```
for( const currentProvider of providersOrder ) {
    avatarImage = await provider[currentProvider](emailHash, targetSize);

    if ( avatarImage ) {
        break;
    }
}
```

**Comment by kokarn** [comment]
Also, no docs update?

File: server.js (hunk 1 above)
Comment (Gyran) [reply server.js]: Dosen't say anything in the spec that the server should lowercase it or that the server should.  Neither does gravatar. But it produces different images for lower/uppercase hashes

File: server.js (hunk 2 above)
Comment (Gyran) [reply server.js]: For me the `while` loop signals the intended behaviour better in this case. As I want to run this loop **until** this condition is met. A for-of loop dosen't signal that is will break in the middle of iterating.

File: server.js (hunk 1 above)
Comment (kokarn) [reply server.js]: Id suggest not lowercasing then in case we want to handle some sort of special casing down the line

File: server.js (hunk 2 above)
Comment (kokarn) [reply server.js]: Cool, let's keep it then

File: server.js (hunk 1 above)
Comment (Gyran) [reply server.js]: yes, we are commenting on an outdated diff

**Review by kokarn — COMMENTED** [review]
Do we not want to be backwards compatible? 👯‍♂

**Comment by Gyran** [comment]
probably, nice catch

**Pull request merged**